test/test_imu_monitor/test_imu_monitor.cpp

```diff
@@ -1,18 +1,77 @@
 #include "Monitors/IMUMonitor.hpp"
 #include <unity.h>
 
-void test_set_valid()
+void test_transition_to_normal_init()
 {
+    // tests the imu will transition to normal state correctly
     IMUMonitor imu_monitor(0);
     sfr::imu::turn_on = true;
     sfr::imu::powered = false;
+    sfr::imu::init_mode = (uint16_t)sensor_init_mode_type::complete;
     imu_monitor.execute();
+    TEST_ASSERT_EQUAL((uint16_t)sensor_mode_type::normal, sfr::imu::mode);
+    TEST_ASSERT_EQUAL(true, sfr::imu::mag_x_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::mag_y_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::mag_z_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::gyro_x_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::gyro_y_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::gyro_z_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::acc_x_average->is_valid());
+    TEST_ASSERT_EQUAL(true, sfr::imu::acc_y_average->is_valid());
+}
+void test_transition_to_abnormal_init()
+{
+    IMUMonitor imu_monitor(0);
+    sfr::imu::turn_on = true;
```

Comment:
These tests look good, but as discussed they just copy the code and ensure the conditions are met. It would be better to change the powered, turn_off, turn_on variables and then call the IMUMonitor execute to ensure the correct cases are being entered. Feel free to send me a slack message if you have any questions!